packages/db/src/loaders/commands/migrate.ts

```diff
@@ -0,0 +1,131 @@
+import { logger } from "@truffle/db/logger";
+const debug = logger("db:loaders:commands:migrate");
+
+import { ContractObject } from "@truffle/contract-schema/spec";
+import { toIdObject, IdObject } from "@truffle/db/meta";
+import { Load } from "@truffle/db/loaders/types";
+
+import { generateContractGet } from "@truffle/db/loaders/resources/contracts";
+import {
+  generateTranasctionNetworkLoad,
+  generateNetworkIdFetch,
+  generateNetworkGet
+} from "@truffle/db/loaders/resources/networks";
+import {
+  LoadableContractInstance,
+  generateContractInstancesLoad
+} from "@truffle/db/loaders/resources/contractInstances";
+
+export interface GenerateMigrateLoadOptions {
+  // we'll need everything for this input other than what's calculated here
+  network: Omit<DataModel.NetworkInput, "networkId" | "historicBlock">;
```

Comment:
I'm a little confused on this one -- you're creating a network type that takes everything from `NetworkInput` and then removes the `networkId` or the `historicBlock`?

Reply:
This says that the `network` option must have every field in `DataModel.NetworkInput`, except `networkId` and `historicBlock` (because those two fields are the ones that `generateMigrateLoad` itself supplies)